File: Interview_Prep_App/native_java/retrofit/src/main/java/com/dreams/interviewprepapp/network/RetrofitBuilders.java
package com.dreams.interviewprepapp.network;

import com.dreams.interviewprepapp.repositories.tc2rgithubrepository.source.remote.Tc2rGithubService;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitBuilders {
    private static final String HOST = "raw.githubusercontent.com";
    private static final String PROTOCOL = "https";
    private static final String PATH =
            "/Tc2r1/DevInterview_Questions/";

    private final OkHttpProvider okHttpProvider = new OkHttpProvider();
    private final okhttp3.OkHttpClient buildClient = okHttpProvider.getClient();


    // In order to send out network requests to an API, I use the Retrofit builder
    // class and specify the base URL for the service.
    // I specify a factory for deserializing the response using the Gson library.
    private final Retrofit buildRetrofit = new Retrofit.Builder()
            .addConverterFactory(GsonConverterFactory.create())
            .client(buildClient)
            .baseUrl(String.format("%s://%s%s", PROTOCOL, HOST, PATH))
            .build();



    public Tc2rGithubService tc2rGithubService =
            buildRetrofit.create(Tc2rGithubService.class);
}

File: Interview_Prep_App/native_kotlin/retrofit/src/main/java/com/interviewprep/kotlinretrofit/network/RetrofitBuilders.kt
package com.interviewprep.kotlinretrofit.network

import com.interviewprep.kotlinretrofit.repository.tc2rgithubrepository.source.remote.Tc2rGithubService
import retrofit2.Retrofit.Builder
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitBuilders {
    companion object {
        private const val BASE_URL = "https://raw.githubusercontent.com/"
    }

    private val okHttpProvider = OkHttpProvider()
    private val buildClient = okHttpProvider.client

    // Fetches JSON response. Create Retrofit Builder using Gson converter factory & give base url of web service.
    // In order to send out network requests to an API, I use the Retrofit builder
    // class and specify the base URL for the service.
    // I specify a factory for deserializing the response using the Gson library.
    private val buildRetrofit = Builder()
        .client(buildClient)
        .addConverterFactory(GsonConverterFactory.create())
        .baseUrl(BASE_URL)
        .build()

    internal var tc2rGithubService = buildRetrofit.create(Tc2rGithubService::class.java)
}
